python/lsst/obs/base/formatters/fitsExposure.py

```diff
@@ -523,6 +524,7 @@ class FitsExposureFormatter(FitsMaskedImageFormatter):
 
     can_read_from_uri = True
     ReaderClass = ExposureFitsReader
+    _cached_fits: tuple[uuid.UUID | None, MemFileManager | None] = (None, None)
```

Comment:
I wonder if it'd be worth it to make this a `threading.local` instance so some (hypothetical) service doing multithreaded reads doesn't thrash the cache.

Reply:
I did wonder whether it would be better to cache the `Exposure` rather than the FITS bytes given that there seems to be real overhead in ExposureFitsReader.

Reply:
Did the overhead appear to be in `ExposureFitsReader` construction?  If not, I think it would make sense to cache that, as that still lets you only read components as needed.